shell/platform/android/io/flutter/plugin/platform/PlatformPlugin.java

```diff
@@ -277,7 +279,11 @@ private void setSystemChromeSystemUIOverlayStyle(
   }
 
   private void popSystemNavigator() {
-    activity.finish();
+    if (activity instanceof OnBackPressedDispatcherOwner) {
+      ((ComponentActivity) activity).getOnBackPressedDispatcher().onBackPressed();
```

Comment:
This is a problematic cast. Just cast directly to `OnBackPressedDispatcherOwner`